# Disables annotating Ruby's declarations as `__declspec(dllimport)` when using them inside of the mkxp-z codebase,
# since our executable is statically linked against Ruby.
# Failure to do this results in Ruby gems failing to load.

--- a/include/ruby/internal/dllexport.h
+++ b/include/ruby/internal/dllexport.h
@@ -39,7 +39,7 @@
 #undef RUBY_EXTERN
 #if defined(MJIT_HEADER) && defined(_WIN32)
 # define RUBY_EXTERN extern __declspec(dllimport)
-#elif defined(RUBY_EXPORT)
+#elif defined(RUBY_EXPORT) || defined(MKXPZ_VERSION)
 # define RUBY_EXTERN extern
 #elif defined(_WIN32)
 # define RUBY_EXTERN extern __declspec(dllimport)
